tests/unit/models/keras/test_utils.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 The Bellman Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest
+import tensorflow as tf
+
+from tests.util.misc import ShapeLike, empty_dataset, random_seed
+from trieste.data import Dataset
+from trieste.models.keras.utils import get_tensor_spec_from_data, sample_with_replacement
+
+
+def test_get_tensor_spec_from_data_raises_for_incorrect_dataset() -> None:
+
+    dataset = empty_dataset([1], [1])
+
+    with pytest.raises(ValueError):
+        get_tensor_spec_from_data(dataset.query_points)
+
+
+@pytest.mark.parametrize(
+    "query_point_shape, observation_shape", [([1], [1]), ([2], [1]), ([5], [1]), ([5], [2])]
+)
+def test_get_tensor_spec_from_data(
+    query_point_shape: ShapeLike, observation_shape: ShapeLike
+) -> None:
+    dataset = empty_dataset(query_point_shape, observation_shape)
+    input_spec, output_spec = get_tensor_spec_from_data(dataset)
+
+    assert input_spec.shape == query_point_shape
+    assert input_spec.dtype == dataset.query_points.dtype
+    assert input_spec.name == "query_points"
+
+    assert output_spec.shape == observation_shape
+    assert output_spec.dtype == dataset.observations.dtype
+    assert output_spec.name == "observations"
+
+
+def test_sample_with_replacement_raises_for_invalid_dataset() -> None:
+
+    dataset = empty_dataset([1], [1])
+
+    with pytest.raises(ValueError):
+        sample_with_replacement(dataset.query_points)
+
+
+def test_sample_with_replacement_raises_for_empty_dataset() -> None:
+
+    dataset = empty_dataset([1], [1])
+
+    with pytest.raises(tf.errors.InvalidArgumentError):
+        sample_with_replacement(dataset)
+
+
+@random_seed
+def test_sample_with_replacement_seems_correct() -> None:
+
+    n_rows = 100
+    x = tf.constant(np.arange(0, n_rows, 1), shape=[n_rows, 1])
```

Comment:
worth checking with x in dim higher than 1? (see other comment below)

Reply:
yep, will do